src/module/data/pseudo-documents/advancements/item-grant-advancement.mjs

```diff
@@ -142,4 +142,92 @@ export default class ItemGrantAdvancement extends BaseAdvancement {
 
     return { [path]: uuids.filter(_ => _) };
   }
+
+  /* -------------------------------------------------- */
+
+  /** @inheritdoc */
+  async reconfigure() {
+    await super.reconfigure();
+
+    const actor = this.document.parent;
+
+    const allowed = await ds.applications.api.DSDialog.confirm({
+      window: {
+        icon: "fa-solid fa-arrow-rotate-right",
+        title: "DRAW_STEEL.ADVANCEMENT.Reconfigure.ConfirmItemGrant.Title",
+      },
+      content: `<p>${game.i18n.localize("DRAW_STEEL.ADVANCEMENT.Reconfigure.ConfirmItemGrant.Content")}</p>`,
+    });
+    if (!allowed) return;
+    const chains = [await ds.utils.AdvancementChain.create(this)];
+    const configuration = await ds.applications.apps.advancement.ChainConfigurationDialog.create({
+      chains, actor, window: { title: "DRAW_STEEL.ADVANCEMENT.ChainConfiguration.reconfigureTitle" },
+    });
+    if (!configuration) return;
+    const toDelete = this.grantedItemsChain().map(i => i.id);
+    if (toDelete.size) await actor.deleteEmbeddedDocuments("Item", Array.from(toDelete));
+
+    // This has a lot of reused logic from AdvancementModel#applyAdvancements, we should investigate if there's
+    // a good way functionalize it for reuse.
```

Comment:
Yeah a refactor would be good. Prefer that is in a follow-up so this PR can focus on just the basic functions.